src/components/canvas.js

```diff
@@ -26,24 +28,75 @@ const canvasView = widget((update) => {
   function onupdate (_render) {
     render = _render
     if (ctx) {
-      render(ctx, canvas.width, canvas.height)
+      render(ctx, canvas.width, canvas.height, changed)
```

Comment:
I think transformed should be applied to the canvas here directly so the renderer doesn't need to know about this information